lib/ecto/repo/transaction.ex

```diff
@@ -16,9 +16,19 @@ defmodule Ecto.Repo.Transaction do
     return = &adapter.rollback(adapter_meta, &1)
 
     case Ecto.Multi.__apply__(multi, repo, wrap, return) do
-      {:ok, values} -> {:ok, values}
-      {:error, {key, error_value, values}} -> {:error, key, error_value, values}
-      {:error, operation} -> raise "operation #{inspect operation} is manually rolling back, which is not supported by Ecto.Multi"
+      {:ok, values} -> 
+        {:ok, values}
+
+      {:error, {key, error_value, values}} -> 
+        {:error, key, error_value, values}
+
+      {:error, operation} -> 
+        raise """
+        operation #{inspect operation} is manually rolling back, which is not supported by Ecto.Multi.
+
+        This can occur if you try to commit an outer transaction when the inner transaction has failed 
+        (and rolled back). Instead, allow the outer transaction to fail as well and handle the error case.
+        """
```

Comment:
```suggestion
      {:error, operation} -> 
        raise """
        operation #{inspect operation} is rolling back unexpectedly . 
        
        This can happen if `repo.rollback/1` is manually called, which is not supported by `Ecto.Multi`. It can also occur if a nested transaction has rolled back and its error is not bubbled up to the outer multi. Nested transactions are discouraged when using `Ecto.Multi`. Consider flattening out the transaction instead. 
        """
```

@v0idpwn / @josevalim  Wanted to see if I could close this one. How does this look to you?

Reply:
This looks good to me

Reply:
Looks good to me. But please break it into multiple lines and have `\` at the end of each line so we remove the line breaks from when the exception is effectively shown. :) Thanks :heart: